packages/connect-core/src/entities/Application.ts

```diff
@@ -24,71 +24,35 @@ export interface ApplicationData {
   repoAddress?: string
   version?: string
 }
-
 export default class Application extends CoreEntity implements ApplicationData {
-  readonly abi?: Abi
   readonly address!: string
   readonly appId!: string
-  readonly appName?: string
-  readonly author?: string
   readonly codeAddress!: string
   readonly contentUri?: string
-  readonly contractPath?: string
-  readonly deprecatedIntents?: { [version: string]: AppIntent[] }
-  readonly description?: string
-  readonly htmlUrl?: string
-  readonly intents?: AppIntent[]
-  readonly icons?: { src: string; sizes: string }[]
   readonly isForwarder?: boolean
   readonly isUpgradeable?: boolean
   readonly kernelAddress!: string
   readonly name?: string
   readonly registryAddress!: string
   readonly registry?: string
   readonly repoAddress?: string
-  readonly sourceUrl?: string
   readonly version?: string
+  #artifact?: string | null
+  #manifest?: string | null
+  abi?: Abi
+  appName?: string
+  author?: string
+  contractPath?: string
+  deprecatedIntents?: { [version: string]: AppIntent[] }
+  description?: string
+  icons?: { src: string; sizes: string }[]
+  intents?: AppIntent[]
+  htmlUrl?: string
+  sourceUrl?: string
```

Comment:
Ah I didn’t think about that… I’m wondering if it’s worth using `readonly` at all then?

Reply:
If we are going to use a method `_init()` / `create()` always required probably not worth it. I can't think other alternatives to try to have `readonly` otherwise.